File: app/src/main/java/com/kota/Bahamut/Pages/Model/ClassPageItem.java
package com.kota.Bahamut.Pages.Model;

import com.kota.Bahamut.ListPage.TelnetListPageItem;
import java.util.Stack;

public class ClassPageItem extends TelnetListPageItem {
    private static final int _count = 0;
    private static final Stack<ClassPageItem> _pool = new Stack<>();
    public String Manager = null;
    public int Mode = 0;
    public String Name = null;
    public String Title = null;
    public boolean isDirectory = false;

    private ClassPageItem() {
    }

    public static void release() {
        synchronized (_pool) {
            _pool.clear();
        }
    }

    public static void recycle(ClassPageItem item) {
        synchronized (_pool) {
            _pool.push(item);
        }
    }

    public static ClassPageItem create() {
        ClassPageItem item = null;
        synchronized (_pool) {
            if (_pool.size() > 0) {
                item = _pool.pop();
            }
        }
        if (item == null) {
            return new ClassPageItem();
        }
        return item;
    }

    public void clear() {
        super.clear();
        this.Manager = null;
        this.Name = null;
        this.Title = null;
        this.isDirectory = false;
        this.Mode = 0;
    }

    /* access modifiers changed from: protected */
    protected void finalize() throws Throwable {
        super.finalize();
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/model/ClassPageItem.kt
package com.kota.Bahamut.pages.model

import com.kota.Bahamut.listPage.TelnetListPageItem
import java.util.Stack

class ClassPageItem private constructor() : TelnetListPageItem() {
    @JvmField
    var manager: String = "" // 版主群
    var mode: Int = 0 // 0: 看板分類, 1: 看板列表
    @JvmField
    var name: String = "" // 看板英文名稱
    @JvmField
    var title: String = "" // 看板中文名稱
    @JvmField
    var isDirectory: Boolean = false // 是否為目錄

    override fun clear() {
        super.clear()
        this.manager = ""
        this.name = ""
        this.title = ""
        this.isDirectory = false
        this.mode = 0
    }

    companion object {
        private val _pool = Stack<ClassPageItem>()
        @JvmStatic
        fun release() {
            synchronized(_pool) {
                _pool.clear()
            }
        }

        @JvmStatic
        fun recycle(item: ClassPageItem) {
            synchronized(_pool) {
                _pool.push(item)
            }
        }

        fun create(): ClassPageItem {
            var item: ClassPageItem? = null
            synchronized(_pool) {
                if (_pool.isNotEmpty()) {
                    item = _pool.pop()
                }
            }
            if (item == null) {
                return ClassPageItem()
            }
            return item
        }
    }
}
